freqtrade/optimize/backtesting.py

```diff
@@ -275,8 +275,34 @@ def load_bt_data_detail(self) -> None:
                 if pair not in self.exchange._leverage_tiers:
                     unavailable_pairs.append(pair)
                     continue
-                self.futures_data[pair] = funding_rates_dict[pair].merge(
-                    mark_rates_dict[pair], on='date', how="inner", suffixes=["_fund", "_mark"])
+
+                if (pair in mark_rates_dict
+                        and len(funding_rates_dict[pair]) == 0
+                        and "futures_funding_rate" in self.config):
+                    mark_rates_dict[pair]["open_fund"] = self.config.get('futures_funding_rate')
+                    mark_rates_dict[pair]["close_fund"] = 0.0
+                    mark_rates_dict[pair]["high_fund"] = 0.0
+                    mark_rates_dict[pair]["low_fund"] = 0.0
+                    mark_rates_dict[pair]["volume_fund"] = 0.0
+                    mark_rates_dict[pair].rename(
+                        columns={'open': 'open_mark',
+                                 'close': 'close_mark',
+                                 'high': 'high_mark',
+                                 'low': 'low_mark',
+                                 'volume': 'volume_mark'},
+                        inplace=True)
+
+                    self.futures_data[pair] = mark_rates_dict[pair]
+                else:
+                    if "futures_funding_rate" in self.config:
+                        self.futures_data[pair] = mark_rates_dict[pair].merge(
+                            funding_rates_dict[pair], on='date',
+                            how="outer", suffixes=["_mark", "_fund"]).fillna(
+                                self.config.get('futures_funding_rate'))
```

Comment:
is it intentional that you fill "everything" missing with `futures_funding_rate`?

assuming you have some mark candles missing (but funding-rate rows are there for whatever reason) - this would fill the mark columns with it as well.

while "funding_rate" is oscillating around 0 (usually) - mark prices are in "candle price" - so for btc, somewhere around 30k at the moment - so this could heavily taint results

Reply:
Ah I meant to only fill funding rate. I'll adjust that to the series instead of the dataframe

Reply:
Done